xamarin/Xamarin.Forms
# [Android] remove android specific target

### Description of Change ###

Having a platform specific target
```
/build/MonoAndroid10/Xamarin.Forms.targets
```

Breaks shared projects

### Testing Procedure ###
Install nuget into a shared project and a netstandard project and ensure xaml.g.cs files run.
After installing the nuget close the project
delete all bin/obj folders
reopen the project

### PR Checklist ###
